manager/src/main/java/org/openremote/manager/security/ManagerKeycloakIdentityProvider.java

```diff
@@ -1021,13 +1021,92 @@ public ClientRepresentation createUpdateClient(String realm, ClientRepresentatio
+
+        return getRealms(realmsResource -> realmsResource.realm(realm).identityProviders().importFrom(importData));
+    }
+
+    public void createUpdateIdentityProvider(String realm, String alias, String providerId, Map<String, String> config) {
+        IdentityProviderRepresentation representation = new IdentityProviderRepresentation();
+        representation.setAlias(alias);
+        representation.setProviderId(providerId);
+        representation.setConfig(config);
+
+        getRealms(realmsResource -> {
+            IdentityProvidersResource identityProvidersResource = realmsResource.realm(realm).identityProviders();
+
+            if (identityProvidersResource.findAll().stream().anyMatch(ipr -> alias.equals(ipr.getAlias()))) {
+                identityProvidersResource.get(alias).update(representation);
+            } else {
+                try (Response response = identityProvidersResource.create(representation)) {
+                    if (response.getStatusInfo().getFamily() != Response.Status.Family.SUCCESSFUL) {
+                        LOG.warning("Failed to create identity provider response=" + response.getStatusInfo().getStatusCode() + ": " + representation);
+                        return null;
+                    }
+                }
+            }
+
+            return identityProvidersResource.get(alias).toRepresentation();
+        });
+    }
+
+    public void createUpdateIdentityProviderMapper(String realm, String idpAlias, String mapperName, String idpMapper, Map<String, String> mapperConfig) {
+        getRealms(realmsResource -> {
+            IdentityProviderResource idpResource = realmsResource.realm(realm).identityProviders().get(idpAlias);
+
+            IdentityProviderMapperRepresentation mapper = new IdentityProviderMapperRepresentation();
+            mapper.setName(mapperName);
+            mapper.setConfig(mapperConfig);
+            mapper.setIdentityProviderAlias(idpAlias);
+            mapper.setIdentityProviderMapper(idpMapper);
+
+            Optional<IdentityProviderMapperRepresentation> existingMapper = idpResource.getMappers().stream().filter(m -> mapperName.equals(m.getName())).findFirst();
+            if (existingMapper.isPresent()) {
+                idpResource.update(existingMapper.get().getId(), mapper);
+            } else {
+                try (Response response = idpResource.addMapper(mapper)) {
+                    if (response.getStatusInfo().getFamily() != Response.Status.Family.SUCCESSFUL) {
+                        LOG.warning("Failed to create identity provider mapper '" + mapperName +  "' response=" + response.getStatusInfo().getStatusCode() + ": " + response.getStatusInfo().getReasonPhrase());
+                    }
+                }
+            }
+
+            return null;
+        });
+    }
+
+    public void addAuthenticationExecutionConfig(String realm, String flowAlias, String executionProviderId, String authenticatorAlias, Map<String, String> authenticatorConfig) {
```

Comment:
Same here as L1031, in regards to applying method overloading. (with `AuthenticatorConfigRepresentation`)